src/WebJobs.Script/Config/HostTelemetryInitializer.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.Config
+{
+    internal class HostTelemetryInitializer : ITelemetryInitializer
+    {
+        private const string HostInstanceIDKey = ScriptConstants.LogPropertyHostInstanceIdKey;
+
+        private readonly ScriptJobHostOptions _hostOptions;
+
+        public HostTelemetryInitializer(IOptions<ScriptJobHostOptions> hostOptions)
+        {
+            if (hostOptions == null)
+            {
+                throw new ArgumentNullException(nameof(hostOptions));
+            }
+
+            _hostOptions = hostOptions.Value;
+        }
+
+        public void Initialize(ITelemetry telemetry)
+        {
+            if (_hostOptions == null)
```

Comment:
No need to check this here -- the constructor already makes sure that it's not null.

Reply:
The constructor only checks if the `IOptions` is null or not. I suppose the value in it could still be null, no? I can move this check to the constructor if you prefer that.

Reply:
Yeah, let's check it in the ctor. No sense in needing to check it every time after that.